metals/src/main/scala/scala/meta/internal/metals/debug/SetBreakpointsRequestHandler.scala

```diff
@@ -31,44 +31,42 @@ private[debug] final class SetBreakpointsRequestHandler(
 
     val originalSource = DebugProtocol.copy(request.getSource)
 
-    val breakPointSymbols = path.toLanguage match {
-      case Language.JAVA =>
-        val topLevels = Mtags.allToplevels(path.toInput)
-        request.getBreakpoints.map { breakpoint =>
-          val symbol = topLevels.occurrences.minBy(distanceFrom(breakpoint))
-          Future.successful(
+    val symbols: Array[(SourceBreakpoint, Option[String])] =
+      path.toLanguage match {
+        case Language.JAVA =>
+          val topLevels = Mtags.allToplevels(path.toInput)
+          request.getBreakpoints.map { breakpoint =>
+            val symbol = topLevels.occurrences.minBy(distanceFrom(breakpoint))
             (breakpoint, Option(JvmSignatures.toTypeSignature(symbol).value))
-          )
-        }
-      case _ =>
-        request.getBreakpoints.map { breakpoint =>
-          val pos = new Position(breakpoint.getLine(), breakpoint.getColumn())
-          compilers.enclosingClass(pos, path).map(sym => (breakpoint, sym))
-        }
-    }
-
-    Future.sequence(breakPointSymbols.toList).flatMap { symbols =>
-      val groups = symbols.groupBy(_._2)
-      val partitions = groups.flatMap {
-        case (Some(fqcn), breakpoints) =>
-          Some(createPartition(request, fqcn, breakpoints.map(_._1).toArray))
-        case _ => None
+          }
+        case _ =>
+          request.getBreakpoints.map { breakpoint =>
+            val pos = new Position(breakpoint.getLine(), breakpoint.getColumn())
+            val sym = classFinder.findClass(path, pos)
+            (breakpoint, sym)
+          }
       }
 
-      val allUris = partitions.map(_.getSource().getPath()).toSet
-      val removed = previousBreakpointUris.diff(allUris)
-      previousBreakpointUris = allUris
+    val groups = symbols.groupBy(_._2)
+    val partitions = groups.flatMap {
```

Comment:
Also, this may be me just not knowing a whole log about DAP stuff, but why are these called `partitions` when these are `Iterable[SetBreakPointsArguments]`?

Reply:
We partition breakpoints into groups that pertain to a certain classfile. So for example if we have:
```scala
class Main{
   ...
}

object Main{
  ...
}
```

Breakpoints from `Main.scala` will be split into partitions for `Main$.class` and `Main.class`

Reply:
Ach and each `SetBreakPointsArguments` is an actual parition.